flask/views.py

```diff
@@ -81,7 +81,7 @@ def as_view(cls, name, *class_args, **class_kwargs):
         """
         def view(*args, **kwargs):
             self = view.view_class(*class_args, **class_kwargs)
-            return self.dispatch_request(*args, **kwargs)
+            return self.dispatch_request()
```

Comment:
is is explicitly documented that dispatch functions are called with all url rule arguments,

so the fix would be to have the default implementation accept them in order to improve the error message

removing the **needed** and **documented** arguments is certainly wrong